subsys/sal/common/sid_on_dev_cert/sid_on_dev_cert.c

```diff
@@ -337,7 +341,23 @@ sid_error_t sid_on_dev_cert_generate_csr(enum sid_on_dev_cert_algo_type algo, ui
 	}
 
 	// Generate key pair
-	if ((ret = sid_pal_crypto_ecc_key_gen(&key_params)) == SID_ERROR_NONE) {
+#ifdef CONFIG_SIDEWALK_CRYPTO_PSA_KEY_STORAGE
+	ret = SID_ERROR_GENERIC;
+	psa_key_id_t key_id =
+		(algo == SID_ODC_CRYPT_ALGO_ED25519) ? SID_CRYPTO_MFG_ED25519_PRIV_KEY_ID :
+		(algo == SID_ODC_CRYPT_ALGO_P256R1)  ? SID_CRYPTO_MFG_SECP_256R1_PRIV_KEY_ID :
+						       PSA_KEY_ID_NULL;
+	int err = sid_crypto_keys_new_generate(key_id, key_params.puk, key_params.puk_size);
+	if (!err) {
+		err = sid_crypto_keys_buffer_set(key_id, key_params.prk, key_params.prk_size);
+		if (!err) {
+			ret = SID_ERROR_NONE;
+		}
+	}
```

Comment:
My proposal to make this snippet simpler
```suggestion
#ifdef CONFIG_SIDEWALK_CRYPTO_PSA_KEY_STORAGE
	psa_key_id_t key_id =
		(algo == SID_ODC_CRYPT_ALGO_ED25519) ? SID_CRYPTO_MFG_ED25519_PRIV_KEY_ID :
		(algo == SID_ODC_CRYPT_ALGO_P256R1)  ? SID_CRYPTO_MFG_SECP_256R1_PRIV_KEY_ID :
						       PSA_KEY_ID_NULL;
	if (sid_crypto_keys_new_generate(key_id, key_params.puk, key_params.puk_size) || sid_crypto_keys_buffer_set(key_id, key_params.prk, key_params.prk_size)) {
		ret = SID_ERROR_GENERIC;
	}
```

Reply:
I left the longer snippet because it's more human readable and more convient to debug